packages/docs/community/release-process/smart-contracts.md

```diff
@@ -37,57 +37,71 @@ When a release is getting proposed on a Baklava/Alfajores/Mainnet, the commit th
-yarn truffle exec --network $NETWORK ./scripts/truffle/verify-bytecode.js --build_artifacts build/rc1 --before_release_1
+yarn verify-deployed -n $NETWORK -b $RELEASE_BRANCH
+```
 
-## Script 2 ##
-# TODO: command to pull source code from current release branch
-# generate report on backwards compatibility between RC1 and current branch with new release contracts
-yarn ts-node scripts/check-backward.ts report --exclude ".*Test|Mock.*|I[A-Z].*|.*Proxy|LinkedList|SortedLinkedList|SortedLinkedListWithMedian|MultiSig.*|ReleaseGold" -o build/rc1 -n build/contracts --output_file report.json
+### Check Backward Compatibility
+
+Compile the contracts in <code>$RELEASE_BRANCH</code> and <code>$BRANCH</code> branches and check backwards compatibility against a release, subject to the following exceptions:
+  1. If the STORAGE version has changed, do not perform backwards compatibility checks
+  2. If the MAJOR version has changed, check storage layout is backwards compatible, but do not check contract ABI is backwards compatible
 
-## Script 3 ##
-# deploy new contracts and generate governance proposal for upgrade
-yarn truffle exec --network development ./scripts/truffle/make-release.js --build_directory build/ --report report.json --proposal proposal.json --initialize_data example-initialize-data.json'
+For changed contracts, confirm that version number in <code>$BRANCH</code> is strictly greater than <code>$RELEASE_BRANCH</code> version number.
+For unchanged contracts, confirm that version number in <code>$BRANCH</code> is exactly the <code>$RELEASE_BRANCH</code> version number.
+Output a JSON backwards compatibility report to <code>$REPORT</code> file.
+
+```bash
+# TODO: command to pull source code from current release branch
+yarn check-versions -a $RELEASE_BRANCH -b $BRANCH -r $REPORT
 ```
 
-These scripts do the following:
+### Deploy New Contracts
+
+Build contracts on <code>$BRANCH</code> and, using corresponding backwards compatibility report <code>$REPORT</code>, deploy changed contracts to <code>$NETWORK</code>.
+STORAGE updates are adopted by deploying a new proxy/implementation pair. Initialize new contracts with values using <code>$INITIALIZE_DATA</code> when specified.
+Output a JSON upgrade governance proposal to <code>$PROPOSAL</code> file. The proposal encodes STORAGE updates by repointing the Registry to the new proxy. Other storage compatible upgrades are encoded by repointing the existing proxy's implementation.
 
-#### Script 1:
+```bash
+yarn make-release -r $REPORT -b $BRANCH -n $NETWORK -i $INITIALIZE_DATA -p $PROPOSAL
+```
 
-1. Compiles the contracts at <code>rc1</code> branch and confirms that the compiled bytecode matches what is currently deployed on the specified network.
+### Submit Upgrade Proposal
 
-#### Script 2:
+Submit the autogenerated upgrade proposal to the Governance contract for review by voters, outputting a unique identifier.
```

Comment:
Should stakeholders also verify the current deployment from the last audited commit